src/mqtt_client.rs

```diff
@@ -202,6 +203,24 @@ impl<
         Ok(())
     }
 
+    /// Specify the authentication message used by the server.
+    ///
+    /// # Args
+    /// * `user_name` - The user name
+    /// * `password` - The password
+    pub fn set_auth(
```

Comment:
I'm hesitant about this because it lets a person provide credentials without enforcing the usage of TLS. These parameters should probably be only accepted IFF we are on the TLS port of MQTT. It would be a big security vulnerability to allow people to transmit plaintext auth credentials over the unsecure MQTT port

Reply:
Maybe we can move these parameters as an option for setting the MQTT secured port?

Reply:
I understand your concern. However, a lot of clients allow people to provide credentials without enforcing the usage of TLS, like [paho-mqtt](https://pypi.org/project/paho-mqtt/). And sometimes people have to connect to a broker that need  credentials. I think maybe we can turn it as a feature and disabled by default. For users who need this function, they can enable this feature anyway.

Reply:
That's.... fair. I just don't think it's a great idea to give engineers a loaded gun pointed directly at their foot. Sure, some may knowingly be doing this because they don't care about keeping the credentials secret, but others may be oblivious to the massive cybersecurity vulnerability that this exposes.

Reply:
In any case, we should have the ability to set authentication credentials on an unsecure channel **disabled by default**. I'm fine with allowing people to opt-in if they acknowledge the risk. There _are_ real world use cases where this makes sense, i.e. if you have a private LAN.

Reply:
I just add the "unsecure" feature to the "set_auth" function so that it is disabled by default